plat/ti/k3/common/bl31.c

```diff
@@ -22,6 +47,34 @@ void bl31_early_platform_setup(bl31_params_t *from_bl2,
 	/* There are no parameters from BL2 if BL31 is a reset vector */
 	assert(from_bl2 == NULL);
 	assert(plat_params_from_bl2 == NULL);
+
+#ifdef PRELOADED_BL32_BASE
+	/* Populate entry point information for BL32 */
+	SET_PARAM_HEAD(&bl32_image_ep_info, PARAM_EP, VERSION_1, 0);
+	bl32_image_ep_info.pc = PRELOADED_BL32_BASE;
```

Comment:
Hmm, it seems BL32_BASE can be directly used as the macro. This can be passed in from the platform.mk based on whether the SPD build flag is set.

It is better to avoid introducing generic build macros if possible.

Reply:
The same could be done with BL33_BASE, but it seems the PRELOADED_ version of that was introduced to avoid confusion between the original which is a load address used by BL2 and earlier stages to put the BL33 at, and the address "preloaded" by an external bootloader. So I would like to keep the same naming here as our BL32 is externally loaded. If at some-point we want to have TF-A load BL32 then we will not define this and BL2 will do the loading with the other address.

Reply:
By the same reasoning, why did BL33_BASE need the PRELOADED_ prefix, it could have also just been "BL33_BASE"? Should we change this? I just want to keep things consistent.

Reply:
Sorry I didn't elaborate previously. PRELOADED_BL33_BASE was actually meant to indicate to the BL2 (the loader) that BL33 doesn't need loading as it already pre-loaded. So its pre-loaded from BL2 point of view. From BL31 point of view, all the images are pre-loaded and this macro came in handy to override the default Non Secure jump address.

Hence as long as the image does not appear pre-loaded to BL2, it is better to avoid using the PRELOADED_ prefix. BL31 can use the BL32_BASE macro to figure out the BL32 load address.

Thinking more about it, PRELOADED_HW_CONFIG_BASE should be just HW_CONFIG_BASE going by the above reasoning. Also this macro doesn't do anything in generic code. Hence this macro should remain platform specific. Please keep the `ARM_PRELOADED_DTB_BASE` macro[1] and and you can introduce a `TI_K3_*` option.  Sorry for the flip flop on this.

[1] At our end, we can re-name this macro to be more consistent with what I mention and perhaps change it to `ARM_HW_CONFIG_BASE`.  We are less conservative with platform specific build option.

Reply:
Then in our case both BL32 and HW_CONFIG should not be loaded by BL2 and so should *have* the preload prefix. Although we don't use BL2 right now, there may be some need for it at some point.

Anyway, this can all be changed later so I'll change them for now to match your preference.